src/main/java/com/spotify/autoscaler/AutoscaleJob.java

```diff
@@ -361,25 +259,39 @@ void run(
     autoscalerMetrics.registerClusterDataMetrics(cluster, currentNodes, database);
     autoscalerMetrics.markClusterCheck();
 
-    if (isTooEarlyToFetchMetrics(cluster, timeSupplier)) {
-      final int newNodeCount = sizeConstraints(cluster, clusterResizeLogBuilder, currentNodes);
+    if (isTooEarlyToAutoscale(cluster, timeSupplier)) {
+      LOGGER.info("Too early to autoscale");
+      final int newNodeCount =
+          nodeCountSettingsConstraints(cluster, clusterResizeLogBuilder, currentNodes)
+              .getDesiredNodeCount();
       if (newNodeCount == currentNodes) {
-        LOGGER.info("Too early to autoscale");
         return;
       } else {
+        LOGGER.info("Ensuring node count within boundaries.");
         updateNodeCount(
             session, cluster, timeSupplier, clusterResizeLogBuilder, newNodeCount, currentNodes);
         return;
       }
     }
     final Duration samplingDuration = getSamplingDuration(cluster, timeSupplier);
-    int newNodeCount =
-        cpuStrategy(cluster, clusterResizeLogBuilder, samplingDuration, currentNodes);
-    newNodeCount =
-        storageConstraints(
-            cluster, clusterResizeLogBuilder, samplingDuration, newNodeCount, currentNodes);
+
+    ScalingEvent newScalingEvent =
```

Comment:
This looks kinda awfull, but not sure how to improve it :-(

Reply:
:(